### PR TITLE
refactor linePlotTests

### LINKED ISSUE #2632: [Testing Audit] Area and Line

### PR DESCRIPTION
**Pull request by ztsai**
closes #2632 

- removed a few unnecessary/redundant tests
- refactored `assertLinePathEqualToDataPoints` so that it works for vertical/horizontal lines too
- restructure the test so there's only one `linePlot` describe block

screen shot 2015-10-19 at 4 22 06 pm

### CODE REVIEW

File: test/plots/linePlotTests.ts [hunk 1]
```diff
@@ -863,10 +674,10 @@ describe("Plots", () => {
         let data = [
           {x: -100, y: -1}, // last element in previous bucket
           {x: 0, y: 2}, // first element in current bucket
-          {x: 0.5, y: 1.5}, // the point to be removed 
+          {x: 0.5, y: 1.5}, // the point to be removed
           {x: 1, y: 1}, // minimum y in current bucket
           {x: 2, y: 4}, // maximum y in current bucket
-          {x: 3, y: 3}, // last elemnt in current bucket 
+          {x: 3, y: 3}, // last elemnt in current bucket
```
Comment (jtlan) [inline test/plots/linePlotTests.ts]: "element"

File: test/testMethods.ts [hunk 2]
```diff
@@ -353,15 +353,30 @@ module TestMethods {
...
-      assert.closeTo(yScale.invert(+coordinates[1]), data[i].y, EPSILON,
-        `Point ${i} drawn, has correct y coordinate`);
+      let command = edge[0];
+      let coordinates = edge.slice(1).split(/,|\s/);
+      switch (command) {
+        case "M": case "m": case "L": case "l":
+          assert.strictEqual(coordinates.length, 2, "There is an x coordinate and a y coordinate");
+          assert.closeTo(xScale.invert(+coordinates[0]), data[i].x, EPSILON,
+            `Point ${i} drawn, has correct x coordinate`);
+          assert.closeTo(yScale.invert(+coordinates[1]), data[i].y, EPSILON,
```
Comment (jtlan) [inline test/testMethods.ts]: Use `TestMethods.assertPointsClose()` here?

File: test/testMethods.ts [hunk 3]
```diff
@@ -353,15 +353,30 @@ module TestMethods {
   export function assertLinePathEqualToDataPoints(path: string, data: {x: number, y: number}[],
     xScale: Plottable.Scales.Linear, yScale: Plottable.Scales.Linear) {
     let EPSILON = 0.0001;
-    let lineEdges = TestMethods.normalizePath(path).match(/(\-?\d+\.?\d*)(,|\s)(-?\d+\.?\d*)/g);
+    let lineEdges = TestMethods.normalizePath(path).match(/([MmLl](\-?\d+\.?\d*)(,|\s)(-?\d+\.?\d*)|([HhVv](\-?\d+\.?\d*)))/g);
```
Comment (jtlan) [inline test/testMethods.ts]: Why this change?

Comment (ztsai) [inline test/testMethods.ts]: moveto(Mm) and lineto(Ll) are followed by (x, y) and horizontal lineto(Hh) and vertical lineto (Vv) are followed by only one coordinate
for step functions, horizontal/vertical lineto are used, so I changed the reg exp to consider those cases too

File: test/plots/linePlotTests.ts [hunk 4]
```diff
@@ -1,412 +1,299 @@
...
+      let dataset: Plottable.Dataset;
+
+      beforeEach(() => {
+        svg = TestMethods.generateSVG();
+        xScale = new Plottable.Scales.Linear();
+        yScale = new Plottable.Scales.Linear();
+        linePlot = new Plottable.Plots.Line<number>();
+        linePlot.x((d: any) => d.x, xScale);
+        linePlot.y((d: any) => d.y, yScale);
+        linePlot.attr("stroke", (d: any, i: number, m: any) => d3.rgb(d.x, d.y, i).toString());
```
Comment (jtlan) [inline test/plots/linePlotTests.ts]: It's kind of odd that this gets set here, when it's only used in a few tests. Can we set it only in the tests where it's used?

File: test/plots/linePlotTests.ts [hunk 5]
```diff
@@ -1,412 +1,299 @@
...
+      beforeEach(() => {
+        svg = TestMethods.generateSVG();
+        xScale = new Plottable.Scales.Linear();
+        yScale = new Plottable.Scales.Linear();
+        dataset = new Plottable.Dataset(data);
+        dataset2 = new Plottable.Dataset(data2);
+        linePlot = new Plottable.Plots.Line<number>();
+        linePlot.x((d) => d.x, xScale);
+        linePlot.y((d) => d.y, yScale);
+        linePlot.attr("stroke", (d, i, m) => d3.rgb(d.x, d.y, i).toString());
```
Comment (jtlan) [inline test/plots/linePlotTests.ts]: Same here -- is this required for the test to work?
